app_feup/lib/controller/networking/network_router.dart

```diff
@@ -177,31 +178,56 @@ class NetworkRouter {
 
   static Future<List<Trip>> getNextArrivalsStop(
       String stopCode, BusStopData stopData) async {
-    final String url =
-        'http://move-me.mobi/NextArrivals/GetScheds?providerName=STCP&stopCode=STCP_' +
-            stopCode;
+
+    final url =
+        'https://www.stcp.pt/pt/itinerarium/soapclient.php?codigo=' + stopCode;
+
     final http.Response response = await http.get(url);
-    final List<Trip> tripList = List();
+    final htmlResponse = parse(response.body);
 
-    final List json = jsonDecode(response.body);
+    final tableEntries =
+        htmlResponse.querySelectorAll('#smsBusResults > tbody > tr.even');
+
+    final configuredBuses = stopData.configuredBuses;
+
+    final tripList = List<Trip>();
+
+    for (var entry in tableEntries) {
+      final info = entry.querySelectorAll('td');
 
-    for (var TripKey in json) {
-      final trip = TripKey['Value'];
-      final String line = trip[0];
-      if (stopData.configuredBuses.contains(line)) {
-        final String destination = trip[1];
-        String timeString = trip[2];
-        if (timeString.substring(timeString.length - 1) == '*') {
-          timeString = timeString.substring(0, timeString.length - 1);
+      final line = info[0].querySelector('ul > li').text.trim();
```

Comment:
Can we change the name of this variable to `busStopsLine`?

Reply:
That is relative to a bus line (or code), not to a bus stop. So perhaps it should be changed to busLine?

Reply:
Sure